x-pack/plugins/maps/public/layers/sources/mvt_single_layer_vector_source/mvt_single_layer_vector_source_editor.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import React, { Fragment, Component, ChangeEvent } from 'react';
+import _ from 'lodash';
+import { EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
+import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
+
+export type MVTSingleLayerVectorSourceConfig = {
+  urlTemplate: string;
+  layerName: string;
+  minSourceZoom: number;
+  maxSourceZoom: number;
+};
+
+export interface Props {
+  onSourceConfigChange: (sourceConfig: MVTSingleLayerVectorSourceConfig) => void;
+}
+
+interface State {
+  urlTemplate: string;
+  layerName: string;
+  minSourceZoom: number;
+  maxSourceZoom: number;
+}
+
+export class MVTSingleLayerVectorSourceEditor extends Component<Props, State> {
+  state = {
+    urlTemplate: '',
+    layerName: '',
+    minSourceZoom: MIN_ZOOM,
+    maxSourceZoom: MAX_ZOOM,
+  };
+
+  _sourceConfigChange = _.debounce(() => {
+    const canPreview =
+      this.state.urlTemplate.indexOf('{x}') >= 0 &&
+      this.state.urlTemplate.indexOf('{y}') >= 0 &&
+      this.state.urlTemplate.indexOf('{z}') >= 0;
+
+    if (canPreview && this.state.layerName) {
+      this.props.onSourceConfigChange({
+        urlTemplate: this.state.urlTemplate,
+        layerName: this.state.layerName,
+        minSourceZoom: this.state.minSourceZoom,
+        maxSourceZoom: this.state.maxSourceZoom,
+      });
+    }
+  }, 200);
+
+  // @ts-ignore
+  _handleUrlTemplateChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const url = e.target.value;
+    this.setState(
+      {
+        urlTemplate: url,
+      },
+      () => this._sourceConfigChange()
+    );
+  };
+
+  // @ts-ignore
```

Comment:
Can `@ts-ignore` be removed? Otherwise, can you add a comment on what is preventing its removal?